File: Book Studio/matrixbook.client/src/app/models/book.ts
import { Chapter } from './chapter';

export interface Book {
    id?: string;
    title?: string;
    subtitle?: string;
    author?: string;
    authorId?: string;
    content?: string;
    summary?: string;
    category?: string;
    tag?: string;
    categoryIds?: string[];
    tagIds?: string[];
    defaultImageId?: string;
    dateCreated?: Date;
    dateUpdated?: Date;
    textCount?: number;
    rank?: number;
    publishOrder?: number;
    images?: string[];
    imageIds?: string[];
    status?: string;
    year?: string;
    isVisible?: boolean;
    speechService?: string;
    language?: string;
    voiceName?: string;
    wavGenerated?: boolean;
    mp3Generated?: boolean;
    mp4Generated?: boolean;
    srtGenerated?: boolean;
    hide?: boolean;
}

export interface ExportBook extends Book {
    chapters?: Chapter[];
    chapterCount?: number;
    chunkCount?: number;
}

export interface BookViewModel extends Book {
    bookCount?: number;
}

export interface BookInfo {
    id?: string;
    title?: string;
    subtitle?: string;
    author?: string;
    status?: string;
    textCount: number;
    dateUpdated?: Date;
    dateCreated?: Date;
}

File: Book Studio/AudioBookStudio.Models/Data/Book.cs


namespace AudioBookStudio.Models.Data;

public class Book : Entity
{
    public required string Title { get; set; }
    public string? Subtitle { get; set; }
    public string? Summary { get; set; }
    public string? Content { get; set; }
    public string? Author { get; set; }
    public string? AuthorId { get; set; }
    public string? Status { get; set; }
    public List<string> CategoryIds { get; set; } = [];
    public List<string> TagIds { get; set; } = [];
    public string? Year { get; set; }
    public string? DefaultImageId { get; set; }
    public string? Language { get; set; }
    public string? VoiceName { get; set; }
    public string? SpeechService { get; set; } = "Edge";

    public bool WavGenerated { get; set; }
    public bool Mp3Generated { get; set; }
    public bool Mp4Generated { get; set; }
    public bool SrtGenerated { get; set; }
    public bool TextGenerated { get; set; }
    public bool Hide { get; set; }

    public float PublishOrder { get; set; } = 10000;
    public float Rank { get; set; }
    public long TextCount { get; set; }
    public DateTime? DateUpdated { get; set; }
    public DateTime? DateCreated { get; set; }
    public List<string> ImageIds { get; set; } = [];
}